source/common/common/matchers.h

```diff
@@ -155,9 +139,65 @@ class StringMatcherImpl : public ValueMatcher, public StringMatcher {
   }
 
 private:
+  // Type `xds::type::matcher::v3::StringMatcher` doesn't have an extension type, so use function
```

Comment:
As per @wbpcode should we update this as well to keep consistency?

Reply:
Yes, I think we can update that as well (in a separate commit), and then remove some of the template/overloading code that makes it work with the types not being identical.

Reply:
SG.